eng/common/TestResources/Remove-TestResources.ps1

```diff
@@ -124,7 +124,13 @@ if (![string]::IsNullOrWhiteSpace($ServiceDirectory)) {
     $preRemovalScript = Join-Path -Path $root -ChildPath 'remove-test-resources-pre.ps1'
     if (Test-Path $preRemovalScript) {
         Log "Invoking pre resource removal script '$preRemovalScript'"
-        &$preRemovalScript -ResourceGroupName $ResourceGroupName @PSBoundParameters
+        if ($BaseName){
+            &$preRemovalScript -ResourceGroupName $ResourceGroupName @PSBoundParameters
```

Comment:
Rather than call the script in both places, just add `ResourceGroupName` to `$PSBoundParameters` (a hashtable; the `$` is used to reference the variable while `@` splats it). Less branching in code this way.

Reply:
I love that ! coming up